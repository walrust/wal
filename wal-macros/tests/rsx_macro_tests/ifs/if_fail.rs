use wal_macros::rsx;

fn main() {
    rsx! { if };
    rsx! { if true };
    rsx! { if {} };
    rsx! { if 1 {} };
    let x = 1;
    rsx! { if x {} };
    rsx! { if true {} else };
    rsx! { if true {} else if };
    rsx! { if true {} else if true };
    rsx! { if true {} else if {} };
    rsx! { if true {} else if true {} else };
    rsx! { if true { {} } };
    rsx! { if true { () } };
    rsx! { if true { invalid_reference } };
}
