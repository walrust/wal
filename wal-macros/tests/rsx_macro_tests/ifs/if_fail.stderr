error: unexpected end of input, expected an expression
 --> tests/rsx_macro_tests/ifs/if_fail.rs:4:5
  |
4 |     rsx! { if };
  |     ^^^^^^^^^^^^
  |
  = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected curly braces
 --> tests/rsx_macro_tests/ifs/if_fail.rs:5:5
  |
5 |     rsx! { if true };
  |     ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Expected condition for `if` expression, found an empty block
 --> tests/rsx_macro_tests/ifs/if_fail.rs:6:16
  |
6 |     rsx! { if {} };
  |                ^^

error: unexpected end of input, expected curly braces
  --> tests/rsx_macro_tests/ifs/if_fail.rs:10:5
   |
10 |     rsx! { if true {} else };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected an expression
  --> tests/rsx_macro_tests/ifs/if_fail.rs:11:5
   |
11 |     rsx! { if true {} else if };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected curly braces
  --> tests/rsx_macro_tests/ifs/if_fail.rs:12:5
   |
12 |     rsx! { if true {} else if true };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Expected condition for `if` expression, found an empty block
  --> tests/rsx_macro_tests/ifs/if_fail.rs:13:32
   |
13 |     rsx! { if true {} else if {} };
   |                                ^^

error: unexpected end of input, expected curly braces
  --> tests/rsx_macro_tests/ifs/if_fail.rs:14:5
   |
14 |     rsx! { if true {} else if true {} else };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `invalid_reference` in this scope
  --> tests/rsx_macro_tests/ifs/if_fail.rs:17:23
   |
17 |     rsx! { if true { invalid_reference } };
   |                       ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0308]: mismatched types
 --> tests/rsx_macro_tests/ifs/if_fail.rs:7:16
  |
7 |     rsx! { if 1 {} };
  |                ^ expected `bool`, found integer

error[E0308]: mismatched types
 --> tests/rsx_macro_tests/ifs/if_fail.rs:9:16
  |
9 |     rsx! { if x {} };
  |                ^ expected `bool`, found integer

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> tests/rsx_macro_tests/ifs/if_fail.rs:15:23
   |
15 |     rsx! { if true { {} } };
   |                       ^^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = help: the following other types implement trait `From<T>`:
             <VNode as From<T>>
             <VNode as From<VComponent>>
             <VNode as From<VElement>>
             <VNode as From<VList>>
             <VNode as From<VText>>
   = note: required for `()` to implement `ToString`
   = note: required for `VNode` to implement `From<()>`

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> tests/rsx_macro_tests/ifs/if_fail.rs:16:23
   |
16 |     rsx! { if true { () } };
   |                       ^^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = help: the following other types implement trait `From<T>`:
             <VNode as From<T>>
             <VNode as From<VComponent>>
             <VNode as From<VElement>>
             <VNode as From<VList>>
             <VNode as From<VText>>
   = note: required for `()` to implement `ToString`
   = note: required for `VNode` to implement `From<()>`
